### PR TITLE
perf(pruner): senders not written to db in full node

### PR DESCRIPTION
**Pull request by emhane**
Skips sender recovery stage for a full node. Instead, senders are recovered on demand in execution stage.

Execution with sender recovery on demand takes just under 3 hours on high end hw. Saves 158 GiB disk space. Comparing to [public node](https://reth.paradigm.xyz/d/2k8BXz24x/reth?orgId=1&refresh=30s&var-instance=localhost:5006&var-blockchain_tree_action=&var-database_action=).

Screenshot 2024-07-08 at 16 20 06
Screenshot 2024-07-09 at 16 21 49
Screenshot 2024-07-09 at 16 22 04

### CODE REVIEW

**Comment by shekhirin** [comment]
Can we get a comparison between a recent full node Execution stage time, and this branch? Would be nice to understand if it's a matter of minutes of overhead, or hours.

**Review by shekhirin — CHANGES_REQUESTED** [review]

File: crates/node/builder/src/setup.rs [hunk 1]
```diff
@@ -101,6 +101,8 @@ where
 
     let prune_modes = prune_config.map(|prune| prune.segments).unwrap_or_default();
 
+    let is_full_node = prune_modes.receipts.is_some();
```
[the remark this answers is not in the document]
Comment (emhane) [reply crates/node/builder/src/setup.rs:104]: that wasn't discussed afaik, cc @mattsse

**Comment by emhane** [comment]
1.0 benchmarks:
execution: 41h 14m

this branch:
execution: 45h

@shekhirin 

I'd say we include this as a temp fix, until https://github.com/paradigmxyz/reth/issues/9270 is addressed

**Comment by emhane** [comment]
will take too long on regular hw

**Pull request closed without merging**